File: 9-PalindromeNumber/9-PalindromeNumber.java
// Last updated: 2/13/2026, 11:33:59 AM
class Solution {
    public boolean isPalindrome(int x) {
        int n = x;
        int r = 0;
        //int d;
        if(x<0) return false;
        while(x!=0){
            int d = x%10;
            r = r*10+d;
            x = x/10;
        }if(n==r) return true;
        else return false;
    }
}

File: 9-PalindromeNumber/9-PalindromeNumber.c
// Last updated: 2/13/2026, 11:34:03 AM
bool isPalindrome(int x) {
   long int rev = 0;
   int temp, rem = x;
   if(x<0)
   return false;
   while (x!=0)
   {
    temp = x%10;
    rev = rev*10+temp;
    x=x/10;
   }
   return rem==rev;
}